v0.11.10 no longer compiles with `cookies` feature

I believe due to the `cookie_store` upgrading its version.

Here is what happens when you try and compile:
```
Checking reqwest v0.11.10
error[E0271]: type mismatch resolving `<std::option::IntoIter<cookie::Cookie<'_>> as Iterator>::Item == cookie::Cookie<'static>`
   --> /home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/cookie.rs:163:33
    |
163 |         self.0.write().unwrap().store_response_cookies(cookies, url);
    |                                 ^^^^^^^^^^^^^^^^^^^^^^ expected struct `cookie::Cookie`, found a different struct `cookie::Cookie`
    |
    = note: perhaps two different versions of crate `cookie` are being used?
note: required by a bound in `cookie_store::CookieStore::store_response_cookies`
   --> /home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/cookie_store-0.15.2/src/cookie_store.rs:77:47
    |
77  |     pub fn store_response_cookies<I: Iterator<Item = RawCookie<'static>>>(
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `cookie_store::CookieStore::store_response_cookies`

error[E0271]: type mismatch resolving `<[closure@/home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/cookie.rs:170:39: 170:94] as FnOnce<(&HeaderValue,)>>::Output == std::option::Option<cookie::Cookie<'static>>`
   --> /home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/cookie.rs:172:56
    |
172 |         self.0.write().unwrap().store_response_cookies(iter, url);
    |                                 ---------------------- ^^^^ expected struct `cookie::Cookie`, found a different struct `cookie::Cookie`
    |                                 |
    |                                 required by a bound introduced by this call
    |
    = note: expected enum `std::option::Option<cookie::Cookie<'static>>`
               found enum `std::option::Option<cookie::Cookie<'_>>`
    = note: perhaps two different versions of crate `cookie` are being used?
    = note: required because of the requirements on the impl of `Iterator` for `std::iter::FilterMap<&mut dyn Iterator<Item = &HeaderValue>, [closure@/home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/cookie.rs:170:39: 170:94]>`
note: required by a bound in `cookie_store::CookieStore::store_response_cookies`
   --> /home/jason/.cargo/registry/src/github.com-1ecc6299db9ec823/cookie_store-0.15.2/src/cookie_store.rs:77:47
    |
77  |     pub fn store_response_cookies<I: Iterator<Item = RawCookie<'static>>>(
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `cookie_store::CookieStore::store_response_cookies`

For more information about this error, try `rustc --explain E0271`.
error: could not compile `reqwest` due to 2 previous errors
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -122,7 +122,7 @@ rustls-pemfile = { version = "0.3", optional = true }
 
 ## cookies
 cookie_crate = { version = "0.16", package = "cookie", optional = true }
-cookie_store = { version = "0.15", optional = true }
+cookie_store = { version = "0.16", optional = true }
 proc-macro-hack = { version = "0.5.19", optional = true }
 
 ## compression

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -121,19 +121,19 @@ rustls-native-certs = { version = "0.6", optional = true }
 rustls-pemfile = { version = "0.3", optional = true }
 
 ## cookies
-cookie_crate = { version = "0.15", package = "cookie", optional = true }
+cookie_crate = { version = "0.16", package = "cookie", optional = true }
 cookie_store = { version = "0.15", optional = true }
 proc-macro-hack = { version = "0.5.19", optional = true }
 
 ## compression
 async-compression = { version = "0.3.7", default-features = false, features = ["tokio"], optional = true }
-tokio-util = { version = "0.6.0", default-features = false, features = ["codec", "io"], optional = true }
+tokio-util = { version = "0.7.0", default-features = false, features = ["codec", "io"], optional = true }
 
 ## socks
 tokio-socks = { version = "0.5.1", optional = true }
 
 ## trust-dns
-trust-dns-resolver = { version = "0.20", optional = true }
+trust-dns-resolver = { version = "0.21", optional = true }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
 env_logger = "0.8"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -127,7 +127,7 @@ proc-macro-hack = { version = "0.5.19", optional = true }
 
 ## compression
 async-compression = { version = "0.3.7", default-features = false, features = ["tokio"], optional = true }
-tokio-util = { version = "0.7.0", default-features = false, features = ["codec", "io"], optional = true }
+tokio-util = { version = "0.7.1", default-features = false, features = ["codec", "io"], optional = true }
 
 ## socks
 tokio-socks = { version = "0.5.1", optional = true }

